service-commands/scripts/userClockValues.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Find the creator nodes associated with a user and their respective clock values.
+ *
+ * Script usage: node userClockValues.js <size> <outFile>
+ */
+const axios = require('axios')
+const CreatorNode = require('@audius/libs/src/services/creatorNode')
+const { Command } = require('commander')
+
+function commaSeparatedList(value, unusedPrevValue) {
+  return value.split(',')
+}
+
+const program = new Command()
+program
+  .usage('')
+  .option('-h, --handles <handles>', 'Audius handle', commaSeparatedList, [])
+  .option('-i, --user-ids <userIds>', 'Audius user id', commaSeparatedList, [])
+
+// export DISCOVERY_PROVIDER_ENDPOINT="https://discoveryprovider.audius.co"
+const discoveryProviderEndpoint = process.env.DISCOVERY_PROVIDER_ENDPOINT
+
+async function getUserByHandle(handle, discoveryProviderEndpoint) {
+  try {
+    return (
+      await axios({
+        url: `/v1/full/users/handle/${handle}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and handle: ${handle} with ${err}`
+    )
+  }
+}
+
+async function getUserById(userId, discoveryProviderEndpoint) {
+  try {
+    const resp = (
+      await axios({
+        url: `/users?id=${userId}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+
+    if (!resp) {
+      throw new Error(`Failed to find user with userId ${userId}`)
+    }
+
+    return resp
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and user id: ${userId} with ${err}`
+    )
+  }
+}
+async function getClockValues({
+  wallet,
+  creator_node_endpoint: creatorNodeEndpoint
+}) {
+  const primaryCreatorNode = CreatorNode.getPrimary(creatorNodeEndpoint)
+  const secondaryCreatorNodes = CreatorNode.getSecondaries(creatorNodeEndpoint)
+
+  if (!creatorNodeEndpoint) {
+    return {
+      primaryNode: '',
+      primaryClockValue: '',
+      secondaryNodes: [],
+      secondaryClockValues: []
+    }
+  }
+
+  return {
+    primaryNode: primaryCreatorNode,
+    primaryClockValue: await CreatorNode.getClockValue(
+      primaryCreatorNode,
+      wallet
+    ),
+    secondaryNodes: secondaryCreatorNodes,
+    secondaryClockValues: await Promise.all(
+      secondaryCreatorNodes.map(secondaryNode =>
+        CreatorNode.getClockValue(secondaryNode, wallet)
+      )
+    )
+  }
+}
+
+async function run() {
+  try {
+    const { handles, userIds } = parseArgsAndEnv()
+    const userClockValues = await Promise.all([
+      ...handles.map(async handle => {
+        return {
+          ...(await getClockValues(
+            await getUserByHandle(handle, discoveryProviderEndpoint)
```

Comment:
I don't think this block of code actually runs all the requests asynchronously. the inner awaits wait for the fns `getUserByHandle` and `getClockValues` to run first, and then the code will await an array of completed fns. I think what we need to do is generate those fns without the await, and have the `await Promise.all(...)` handle all the requests.

Reply:
also, I think we should consider storing outputs here into variables. it's a little hard to understand what's going on in lines 95-111 with the multiple spread operators and mapping fns